Funcy/FuncyList.cs

```diff
@@ -316,6 +316,28 @@ public override FuncyList<TReturn> ComputeWith<TReturn>(Func<T, FuncyList<TRetur
 
     public static class FuncyListNT
     {
+        public static Maybe<T> ElementAt<T>(this FuncyList<T> source, int index)
+        {
+            if (index < 0)
+            {
+                return Maybe<T>.None();
+            }
+
+            if (index == 0)
+            {
+                return FuncyListNT.TakeFirst<T>(source);
+            }
+
+            if (source is Nil<T>)
+            {
+                return Maybe<T>.None();
+            }
+
+            Cons<T> cons = source.ToCons();
+
+            return FuncyListNT.ElementAt<T>(cons.Tail, index - 1);
```

Comment:
Yes, you are right.
I considered a cost for object creation, but not for recursive call. Iterative process is prefer, especially Funcy inside.